### DISCUSSION TITLE
Is there a way to implement a transformer for TRPC?

### DISCUSSION THREAD

**Discussion by tothdanielax** [sinclairzx81/typebox#778] [Q&A]
Is there a similar way to implement a TypeBox based tranformer (for decoding/encoding MongoId _ids mainly) for TRPC?

```js
import { uneval } from 'devalue';
import superjson from 'superjson';

// [...]

export const transformer = {
  input: superjson,
  output: {
    serialize: (object) => uneval(object),
    // This `eval` only ever happens on the **client**
    deserialize: (object) => eval(`(${object})`),
  },
};
```

**Comment by sinclairzx81** [comment]
@tothdanielax Hi,

I'm not familiar with tRPC transformers, but TypeBox does support both data encoding and decoding, so I expect it should be technically possible. 

#### Relevant TypeBox Documentation

https://github.com/sinclairzx81/typebox#types-transform
https://github.com/sinclairzx81/typebox#values-decode
https://github.com/sinclairzx81/typebox#values-decode

I'm not sure how these functions would integrate with tRPC at a high level, but would be interested in community experimentation to investigate how a an integration could work (and identify any integration issues). Would you like to investigate a possible integration? Would be happy to have TypeBox reference a community implementation (both default tRPC type integration and transformers), as quite a few users have asked about tRpc integration in the past.
